pkg/restapi/operation/operations.go

```diff
@@ -134,13 +136,29 @@ func (c *Operation) createDataVault(rw http.ResponseWriter, config *models.DataV
 		return
 	}
 
-	err := c.vaultCollection.createDataVault(config.ReferenceID)
+	// Creates data vault config store if it does not already exist to store the dataVaultConfiguration
+	err := c.vaultCollection.createConfigVault()
 	if err != nil {
 		writeCreateDataVaultFailure(rw, err, configBytesForLog)
 		return
 	}
 
-	writeCreateDataVaultSuccess(rw, config.ReferenceID, hostURL, configBytesForLog)
+	vaultID := uuid.New().String()
+	err = c.vaultCollection.storeDataVaultConfiguration(config, vaultID)
+
```

Comment:
Usually we have the `if err != nil` line directly after the call that created the `err`. So you should remove the line break here for consistency.